### PR TITLE
Min python version is now py310

### PR DESCRIPTION
**Pull request by nabobalis**
The whatsnew is updated by https://github.com/sunpy/sunpy/pull/7383/ and not here.

This is in preparation for the pyproject.toml where we need to bump to a min of 310 as well as our support cycle.

### CODE REVIEW

File: sunpy/coordinates/utils.py [hunk 1]
```diff
@@ -420,7 +420,7 @@ def solar_angle_equivalency(observer):
     <Quantity 1.00603718 arcsec>
     """
 
-    if not isinstance(observer, (SkyCoord, BaseCoordinateFrame)):
+    if not isinstance(observer, SkyCoord | BaseCoordinateFrame):
```
Comment (dstansby) [inline sunpy/coordinates/utils.py:423]: 🥳

**Comment by dstansby** [comment]
The figure devdeps fail is:

> E   astropy.utils.exceptions.AstropyDeprecationWarning: Importing ErfaWarning from astropy.utils.exceptions was deprecated in version 6.1 and will stop working in a future version. Instead, please use from erfa import ErfaWarning

Worth fixing here, or for another PR?

**Comment by nabobalis** [comment]
That is fixed in another PR, waiting for Albert to review

**Review by CyclingNinja — APPROVED** [review]

File: sunpy/coordinates/utils.py (hunk 1 above)
Comment (Cadair) [reply sunpy/coordinates/utils.py:423]: Is this actually a good change? It seems amazingly pointless to me.

**Review by Cadair — APPROVED** [review]

File: sunpy/coordinates/utils.py (hunk 1 above)
Comment (CyclingNinja) [reply sunpy/coordinates/utils.py:423]: I think its more readable, its much clearer that a logical operator is happening at that point imo

Comment (nabobalis) [reply sunpy/coordinates/utils.py:423]: THE RUFF SAYS WE DO IT SO WE DO IT

**Comment by nabobalis** [comment]
Oh that reminds me:

```
E       asdf.exceptions.AsdfWarning: asdf.extensions plugin from package asdf-astropy==0.1.1 failed to load:
E       
E       AsdfWarning: asdf.resource_mappings plugin from package asdf==2.8.3 failed to load:
E       
E       RuntimeError: Unable to locate core schemas
```
can we delete asdf?

**Pull request merged**